StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1217UsingStaticDirectivesMustBeOrderedAlphabetically.cs

```diff
@@ -0,0 +1,87 @@
+    /// </para>
+    /// </remarks>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class SA1217UsingStaticDirectivesMustBeOrderedAlphabetically : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The ID for diagnostics produced by the <see cref="SA1217UsingStaticDirectivesMustBeOrderedAlphabetically"/> analyzer.
+        /// </summary>
+        public const string DiagnosticId = "SA1217";
+        private const string Title = "Using static directives must be ordered alphabetically";
+        private const string MessageFormat = "The using static directive for '{0}' must appear after the using static directive for '{1}'";
+        private const string Category = "StyleCop.CSharp.OrderingRules";
+        private const string Description = "All using static directives must be ordered alphabetically.";
+        private const string HelpLink = "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1217.md";
+
+        private static readonly DiagnosticDescriptor Descriptor =
+            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, true, Description, HelpLink);
+
+        private static readonly ImmutableArray<DiagnosticDescriptor> SupportedDiagnosticsValue =
+            ImmutableArray.Create(Descriptor);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => SupportedDiagnosticsValue;
+
+        /// <inheritdoc/>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleCompilationUnit, SyntaxKind.CompilationUnit);
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleNamespaceDeclaration, SyntaxKind.NamespaceDeclaration);
+        }
+
+        private static void HandleCompilationUnit(SyntaxNodeAnalysisContext context)
+        {
+            var compilationUnit = (CompilationUnitSyntax)context.Node;
+            CheckUsingDeclarations(context, compilationUnit.Usings);
+        }
+
+        private static void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var namespaceDirective = (NamespaceDeclarationSyntax)context.Node;
+            CheckUsingDeclarations(context, namespaceDirective.Usings);
+        }
+
+        private static void CheckUsingDeclarations(SyntaxNodeAnalysisContext context, SyntaxList<UsingDirectiveSyntax> usingDirectives)
+        {
+            UsingDirectiveSyntax lastStaticUsingDirective = null;
+
+            foreach (var usingDirective in usingDirectives)
+            {
+                if (usingDirective.StaticKeyword.IsKind(SyntaxKind.StaticKeyword))
+                {
+                    if (lastStaticUsingDirective != null)
+                    {
+                        var firstName = lastStaticUsingDirective.Name.ToFullString();
+                        var secondName = usingDirective.Name.ToFullString();
```

Comment:
Why ToFullString? I dont think ToFullString would work correctly in this example:
```csharp
using static
//a
B;
using static
//b
A;
```